core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -46,7 +46,34 @@ object Automation {
         return rank
     }
 
-
+    // More complicated logic to properly weigh Food vs other Stats (esp Production)
+    private fun getFoodModWeight(city: City, surplusFood: Float): Float {
+        // Zero out Growth if close to Unhappiness limit
+        if (city.civ.getHappiness() < -8) {
+            return 0f
+        } else if (city.civ.isAI()) {
```

Comment:
Sorry for being pedantic, but this is a definite code cleanup issue
All of this can be "unnested" - all the "else" main branches are unnecessary, since the first and second "if"s always return values

Reply:
Fixed. Thanks.